framework/SwiftPlot/Series.swift

```diff
@@ -1,28 +1,38 @@
 // struct defining a data series
-public struct Series {
+public struct Series<T,U> {
     public var barGraphSeriesOptions = BarGraphSeriesOptions()
     public var scatterPlotSeriesOptions = ScatterPlotSeriesOptions()
-    public var points = [Point]()
-    public var scaledPoints = [Point]()
+    public var pairs = [Pair<T,U>]()
+    public var scaledPairs = [Pair<T,U>]()
     public var maxY: Float = 0
     public var minY: Float = 0
     public var label = "Plot"
     public var color : Color = .blue
     public var startColor: Color? = nil
     public var endColor: Color? = nil
     public init() {}
-    public init(points p: [Point], label l: String, color c: Color = Color.lightBlue, hatchPattern: BarGraphSeriesOptions.Hatching = .none, scatterPattern: ScatterPlotSeriesOptions.ScatterPattern = .circle){
-        points = p
+
+    public init(pairs : [Pair<T,U>],
+                label l: String,
```

Comment:
I'd just use label as the parameter name, and drop the l. You can always use `self.label = label` to differentiate between local parameters and class properties.